### DISCUSSION TITLE
Empty cells

### DISCUSSION THREAD

**Discussion by Jerrry7** [casadi/casadi#3273] [Q&A]
Hi,
I am trying to create matrices of size 25 times 1 to include computation results(basically mutliplication and addition) from roughly 50000 parameters, however the matrix is turning out to be empty. Does anyone have an idea what might be wrong?
With regards

**Comment by jgillis** [comment]
Something like MX.zeros(25,1)?

**Reply by Jerrry7** [reply]
I have multiple inputs of form SX.zeros(25,1) and after adding it in matlab it return [], although it is able to show all the inputs individually

**Reply by jgillis** [reply]
Provide some code please

**Reply by Jerrry7** [reply]
[casadi_issue.zip](https://github.com/casadi/casadi/files/12038634/casadi_issue.zip)
This the code that highlight the issue

**Reply by jgillis** [reply]
Running `casadi_problem`,  get 
```
Error using casadi.SparsityInterfaceCommon/vertcat (line 65)
.../casadi/core/sparsity.cpp:1396: Assertion "sp_nrow==ret_nrow || sp_nrow==0" failed:
Sparsity::horzcat: Mismatching number of rows

Error in casadi_problem (line 130)
psi_bar = [psi_hat_1; ..
```

Your code has:
```
psi_bar = [psi_hat_1; ... 
          psi_hat_2;  ...
          psi_hat_3;...
          psi_hat_4...
          psi_hat_5; ... 
          psi_hat_6; ... 
          psi_hat_7;...
          psi_hat_8...
          psi_hat_9; ... 
          psi_hat_10; ... 
          psi_hat_11;...
          psi_hat_12...
          psi_hat_13; ... 
          psi_hat_14; ... 
          psi_hat_15;...
          psi_hat_16...
          psi_hat_17; ... 
          psi_hat_18; ... 
          psi_hat_19;...
         ];
```

With all `psi_hat_*` having size `1-by-1`.
This is a weird construction. Did you miss a semicolon on some lines like after `psi_hat_4`?

In pure Matlab you would get a similar error in fact:
```
Error using vertcat
Dimensions of arrays being concatenated are not consistent.

Error in casadi_problem (line 149)
psi_bar = [psi_hat_1; ...
```

**Reply by Jerrry7** [reply]
Hi, yes I did not put it right, here is the correct construction that highlights the problem.
With regards
[casadi_issue.zip](https://github.com/casadi/casadi/files/12122955/casadi_issue.zip)

rohit

**Reply by jgillis** [reply]
Okay, but this code just runs fine without error?
I just get a complicated SX expression with properties below:
There is a printing overflow happening on Matlab Linux though, making the printout look either garbled or empty.

```matlab
>> class(psi_bar)

ans =

    'casadi.SX'

>> size(psi_bar)

ans =

    19     1

>> n_nodes(psi_bar)

ans =

     2001467

```

**Reply by Jerrry7** [reply]
Is there a way to make sure that this does not create a problem in further code or to print the expression?
